submissions/smillims/oop/app.js

```diff
@@ -0,0 +1,123 @@
+const block = {
+	width: 101,
+	height: 85
+}
+
+// Enemies our player must avoid
+const Enemy = function(x, y, player) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+
+	 this.x = x;
+	 this.y = y;
+	 this.player = player;
+	 this.speed = 150 + (Math.random() * 100); //speed between 100 and 250;
+
+    // The image/sprite for our enemies, this uses
+	 // a helper we've provided to easily load images
+	 
+    this.sprite = 'images/enemy-bug.png';
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function(dt) {
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for
+	 // all computers.
+	 this.x += this.speed * dt;
+	 if (this.x > ctx.canvas.width + block.width){
+		 this.x = -101 * 2;
+	 }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const checkCollisions = () => {
+	allEnemies.forEach(enemy => {
+		if(Math.abs(enemy.x - player.x) < 50 &&
+			Math.abs(enemy.y - player.y) < 50 ||
+			player.y < block.height - 85){
+			player.startOver();
+		}
+	})
+}
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+
+const Player = function(x, y, player) {
+	this.x = x;
+	this.y = y;
+	this.player = player;
```

Comment:
Why do you need `this.player`? It doesn't seem to be used anywhere in `Player`'s methods. Besides if this is a selfreference then `this` is already a selfreference.

Reply:
Yes, it was self reference, so I removed it